src/modules/r.rs

```diff
@@ -0,0 +1,101 @@
+use regex::Regex;
+
+use super::{Context, Module, RootModuleConfig};
+
+use crate::configs::r::RConfig;
+use crate::formatter::StringFormatter;
+
+const R_VERSION_PATTERN: &str = r" (?P<rversion>\d+\.\d+\.\d+) ";
+
+/// Creates a module with the current R programming language version.
+///
+/// Will display the R programming language version if any of the following criteria are met:
+///     - Current directory contains a `.R` file
+pub fn module<'a>(context: &'a Context) -> Option<Module<'a>> {
+    let mut module = context.new_module("r");
+    let config: RConfig = RConfig::try_load(module.config);
+
+    let is_r_project = context
+        .try_begin_scan()?
+        .set_extensions(&["R", "Rproj"])
+        .is_match();
+    if !is_r_project {
+        return None;
+    }
+
+    let r_version = context.exec_cmd("R", &["--version"])?.stderr;
+    let formatted_version = parse_version(&r_version)?;
```

Comment:
Version extraction and formatting should be done lazily, please move it under `version` arm of the formatter variable mapper.